src/utils/router_state.rs

```diff
@@ -0,0 +1,420 @@
+pub const LOADING_AND_TAKEOFF_TIME_MIN: f32 = 10.0;
+/// Time to block vertiport for cargo unloading and landing
+pub const LANDING_AND_UNLOADING_TIME_MIN: f32 = 10.0;
+/// Average speed of cargo aircraft
+pub const AVG_SPEED_KMH: f32 = 60.0;
+
+/// Creates all possible flight plans based on the given request
+/// * `vertiport_depart` - Departure vertiport - svc-storage format
+/// * `vertiport_arrive` - Arrival vertiport - svc-storage format
+/// * `departure_time` - Departure time
+/// * `arrival_time` - Arrival time
+/// * `aircrafts` - Aircrafts serving the route and vertiports
+/// # Returns
+/// A vector of flight plans
+pub fn get_possible_flights(
+    vertiport_depart: Vertiport,
+    vertiport_arrive: Vertiport,
+    departure_time: Option<Timestamp>,
+    arrival_time: Option<Timestamp>,
+    aircrafts: Vec<Vehicle>,
+) -> Result<Vec<FlightPlanData>, String> {
+    info!("Finding possible flights");
+    //1. Find route and cost between requested vertiports
+    info!("[1/5]: Finding route between vertiports");
+    if !is_router_initialized() {
+        error!("Router not initialized");
+        return Err("Router not initialized".to_string());
+    }
+    let (route, cost) = get_route(RouteQuery {
+        from: get_node_by_id(&vertiport_depart.id)?,
+        to: get_node_by_id(&vertiport_arrive.id)?,
+        aircraft: Aircraft::Cargo,
+    })?;
+    debug!("Route: {:?}", route);
+    debug!("Cost: {:?}", cost);
+    if route.is_empty() {
+        error!("No route found");
+        return Err("Route between vertiports not found".to_string());
+    }
+
+    //2. calculate blocking times for each vertiport and aircraft
+    info!("[2/5]: Calculating blocking times");
+    let block_departure_vertiport_minutes = LOADING_AND_TAKEOFF_TIME_MIN;
+    let block_arrival_vertiport_minutes = LANDING_AND_UNLOADING_TIME_MIN;
+    let block_aircraft_minutes = estimate_flight_time_minutes(cost, Aircraft::Cargo);
+    debug!(
+        "Estimated flight time in minutes: {}",
+        block_aircraft_minutes
+    );
+
+    //3. check vertiport schedules and flight plans
+    info!("[3/5]: Checking vertiport schedules and flight plans");
+    const SAMPLE_CAL: &str =
+        "DTSTART:20221020T180000Z;DURATION:PT1H\nRRULE:FREQ=WEEKLY;BYDAY=MO,TU,WE,TH,FR";
+    let departure_vertiport_schedule = Calendar::from_str(SAMPLE_CAL).unwrap(); //TODO get from DB
+    let arrival_vertiport_schedule = Calendar::from_str(SAMPLE_CAL).unwrap(); //TODO get from DB
+
+    if departure_time.is_none() && arrival_time.is_none() {
+        error!("Either departure or arrival time must be specified");
+        return Err("Either departure_time or arrival_time must be set".to_string());
+    }
+
+    let (departure_time, arrival_time) = if departure_time.is_some() {
+        let departure_time = Tz::UTC.from_utc_datetime(
+            &NaiveDateTime::from_timestamp_opt(
+                departure_time.as_ref().unwrap().seconds,
```

Comment:
last comment mentioning unhandled unwrap(), ctrl +f 

offending files are: schedule.rs, router_state.rs, and router.rs

UTs can have unhandled unwraps, e.g. the with_ymd_and_hms calls in schedule.rs

Reply:
This is handled by few lines above as `if departure_time.is_none() && arrival_time.is_none()...` so it's safe to unwrap here